src/services/analytics/useGtm.ts

```diff
@@ -19,7 +19,10 @@ const useGtm = () => {
 
   // Initialize GTM, or clear it if analytics is disabled
   useEffect(() => {
-    isAnalyticsEnabled ? gtmInit() : gtmClear()
+    // router.pathname doesn't contain the safe address
+    // so we can override the initial dataLayer
+    isAnalyticsEnabled ? gtmInit(router.pathname) : gtmClear()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Not the best solution, we should use a state variable here that would be assigned to the initial pathname only once.